### PR TITLE
fixes issue #6832

### LINKED ISSUE #6832: F# compiler should set major version of Embedded PDB
When using `DebugType=embedded` with F# projects, and then trying to load the resulting library with `PEReader`, I am getting an exception. This is because `PEReader` is validating that the version is at least the expected version:

https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEReader.EmbeddedPortablePdb.cs#L52-L70

**Repro steps**

Provide the steps required to reproduce the problem:

1. Set `<DebugType>embedded</DebugType>` in a .NET Standard .fsproj
2. Build
3. Pass the built `.dll` into the following code:

```C#
string filename = ...;
using (var fileStream = File.OpenRead(filename))
{
    PEReader reader = new PEReader(fileStream);

    foreach (var entry in reader.ReadDebugDirectory())
    {
        if (entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb)
        {
            reader.ReadEmbeddedPortablePdbDebugDirectoryData(entry);
        }
    }
}

```

**Expected behavior**

`PEReader` shouldn't throw on the assembly.

**Actual behavior**

```
System.BadImageFormatException
  HResult=0x8007000B
  Message=Unsupported format version: 0.0
  Source=System.Reflection.Metadata
  StackTrace:
   at System.Reflection.PortableExecutable.PEReader.ValidateEmbeddedPortablePdbVersion(DebugDirectoryEntry entry) in F:\workspace\_work\1\s\src\System.Reflection.Metadata\src\System\Reflection\PortableExecutable\PEReader.EmbeddedPortablePdb.cs:line 62
```

**Known workarounds**

Don't use embedded PDBs.

cc @KevinRansom @tmat

### PR DESCRIPTION
**Pull request by KevinRansom**
This fixes #6832 --- The error when reading an F# dll with an embedded pdb file.

The fix is trivial, it just sets the version major field of the Debug Directory entry to 0x100.  The spec was not clear about what this should be.  Tomas has cleared it up.

This PR also embeds a portable pdb in each FSharp product assembly.

The AssemblyVersion validation code has been updated to ensure that the .dll is loadable by SRM and also that the product dll's contain an embedded pdb file.

The AssemblyValidation code is now built as a binary rather than run under fsi.  The reason for this is that, the coreclr contains SRM by default, whereas the desktop does not.

### CODE REVIEW

File: src/absil/ilwritepdb.fs [hunk 1]
```diff
@@ -186,7 +186,7 @@ let pdbGetEmbeddedPdbDebugInfo (embeddedPdbChunk: BinaryChunk) (uncompressedLeng
         Buffer.BlockCopy(stream.ToArray(), 0, buffer, offset, size)
         buffer
     { iddCharacteristics = 0                                                    // Reserved
-      iddMajorVersion = 0                                                       // VersionMajor should be 0
```
Comment (brettfo) [inline src/absil/ilwritepdb.fs]: Comment doesn't align.

**Comment by brettfo** [comment]
ADO is incorrectly reporting the build status.  Real CI is [here](https://dev.azure.com/dnceng/public/_build/results?buildId=277723), but I don't know how to make GH let you merge.

Maybe try rebasing and force pushing?  That'll generate new hashes which may make things respond again.

**Comment by KevinRansom** [comment]
@brettfo who do we need to talk to about the ci being stalled?

**Comment by brettfo** [comment]
From internal email:

> This is unfortunately  a semi-common occurrence which stems from the Azure DevOps GitHub app.  You _might_ get progress from retrying with commenting “/azp run” (I could not as I lack permissions in Fsharp land evidently) but otherwise the workaround would be to recreate the PR.

**Pull request merged**